Examples/Python/src/MagneticField.cpp

```diff
@@ -14,16 +14,84 @@
 #include "Acts/MagneticField/NullBField.hpp"
 #include "Acts/MagneticField/SolenoidBField.hpp"
 #include "Acts/Plugins/Python/Utilities.hpp"
+#include "ActsExamples/MagneticField/FieldMapRootIo.hpp"
+#include "ActsExamples/MagneticField/FieldMapTextIo.hpp"
 
 #include <memory>
 
+#include <boost/filesystem.hpp>
 #include <pybind11/pybind11.h>
 
 namespace py = pybind11;
 using namespace pybind11::literals;
 
-namespace Acts::Python {
+namespace ActsExamples {
+// MagneticFieldMapRz/Xyz are subclasses of InterpolatedMagneticField2/3 used
+// here simply to keep them separate for the Python bindings
+class MagneticFieldMapRz final
```

Comment:
What's the advantage of this class + factory and constructor over a free function that you expose to python in isolation?

Reply:
I tried the free function, but couldn't get it to work at the time because the returned `InterpolatedMagneticField3` was not recognised as a `MagneticFieldProvider`.
I have since discovered that you can specify different base classes in `py::class_<D,B1,B2>` (the `pybind11` docs I saw only mentioned one base class). I subsequently used that for my `MagneticFieldMapXyz` so it could also be used by `RootBFieldWriter` (requires an `InterpolatedMagneticField`), so maybe the same trick would work for `InterpolatedMagneticField3`.
Do you think that would be better?

Reply:
It seems to work everywhere, so I committed this change.
The PR is still WiP because I get very bad physics performance results, which I'm investigating.